src/routing/core/messaging/handling/mod.rs

```diff
@@ -94,44 +98,53 @@ impl Core {
         sender: SocketAddr,
         message: SectionInfoMsg,
         dest_info: DestInfo, // The DestInfo contains the XorName of the sender and a random PK during the initial SectionQuery,
+        our_local_ip: IpAddr,
     ) -> Vec<Command> {
         // Provide our PK as the dest PK, only redundant as the message itself contains details regarding relocation/registration.
         let dest_info = DestInfo {
             dest: dest_info.dest,
             dest_section_pk: *self.section().chain().last_key(),
         };
         match message {
-            SectionInfoMsg::GetSectionQuery(pk) => {
-                let name = XorName::from(pk);
-                debug!("Received GetSectionQuery({}) from {}", name, sender);
-
-                let response = if let (true, Ok(pk_set)) =
-                    (self.section.prefix().matches(&name), self.public_key_set())
+            SectionInfoMsg::GetSectionQuery {
+                public_key,
+                is_node,
+            } => {
+                let name = XorName::from(public_key);
+                let response = if is_node && comm::is_reachable(our_local_ip, &sender).await.is_ok()
```

Comment:
I'm getting a weird error I think due to this call to `is_reachable` as it seems something becomes not `Send`

Reply:
I was able to fix the compilation issue by changing `max_recursion` in sn_node.